drivers/homeless_summary_report/app/controllers/homeless_summary_report/warehouse_reports/reports_controller.rb

```diff
@@ -0,0 +1,169 @@
+###
+# Copyright 2016 - 2021 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module HomelessSummaryReport::WarehouseReports
+  class ReportsController < ApplicationController
+    include WarehouseReportAuthorization
+    include AjaxModalRails::Controller
+    include ArelHelper
+    include BaseFilters
+
+    before_action :require_can_access_some_version_of_clients!, only: [:details]
+    before_action :set_report, only: [:show, :destroy, :details]
+    before_action :set_variants, only: [:show, :details]
+    before_action :set_m1_fields, only: [:show, :details]
+    before_action :set_m2_fields, only: [:show]
+    before_action :set_m7_fields, only: [:show, :details]
+
+    def index
+      @reports = report_scope.ordered.
+        page(params[:page]).per(25)
+      @report = report_class.new(user_id: current_user.id)
+      # Make sure the form will work
+      filters
+    end
+
+    def show
+      @measures = {
+        'Measure 1': @m1_fields.keys,
+        'Measure 2': @m2_fields,
+        'Measure 7': @m7_fields,
+      }
+    end
+
+    def create
+      @report = report_class.new(
+        user_id: current_user.id,
+      )
+      @report.filter = @filter
+      @report.save
+      # TODO: Now we need to figure out how to define the job. I think it's by modifying the report class, probably ultimately the run_and_save! method. Yes.
+      ::WarehouseReports::GenericReportJob.perform_later(
+        user_id: current_user.id,
+        report_class: @report.class.name,
+        report_id: @report.id,
+      )
+      # Make sure the form will work
+      filters
+      respond_with(@report, location: homeless_summary_report_warehouse_reports_reports_path)
+    end
+
+    def destroy
+      @report.destroy
+      respond_with(@report, location: homeless_summary_report_warehouse_reports_reports_path)
+    end
+
+    def details
+      @variant = details_params['variant'] || 'all_persons'
+      @cell = "spm_#{details_params['cell']}" || 'spm_m1a_es_sh_days'
+      @cell_name = @cell.humanize.split(' ')[1..].join(' ')
+      if details_params['cell'].start_with?('m1')
+        @measure = 'Measure 1'
+        @data_cells = @m1_fields.keys
+      elsif details_params['cell'].start_with?('m2')
+        @measure = 'Measure 2'
+        @data_cells = ['m2_reentry_days']
+      elsif details_params['cell'].start_with?('m7')
+        @measure = 'Measure 2'
+        @data_cells = @m7_fields
+      end
+      @detail_clients = @report.clients.send(@variant).send(@cell)
+      @spm_id = @detail_clients.first.send("spm_#{@variant}")
```

Comment:
All of this setup should be pushed down to the model.  Setup a method that accepts `variant` and `cell`, pass in `details_params['variant']` and `details_params['cell']`, return a hash or similar with the bits needed in the details views.